### DISCUSSION TITLE
Request for JavaScript Script to Detect Pathname Changes and Reloads Across Frameworks

### DISCUSSION THREAD

**Discussion by basit-qayoom** [vercel/next.js#68656] [Help]
### Summary

I’m looking for a JavaScript solution to detect pathname changes and reloads across various frameworks and setups. The goal is to handle pathname changes and print the pathname whenever it changes or when the page is reloaded.

Here’s what I’m aiming to achieve:

1. Detect and log pathname on page reload: When the page is refreshed or reloaded, the current pathname should be logged.
2. Detect and log pathname on route changes: When navigating to a different route, the new pathname should be logged.

The behavior should be similar to:
```
const pathname = usePathname();
useEffect(() => {
  console.log(pathname);
}, [pathname]);
```

However, I need a solution that works across different frameworks and setups without relying on specific framework features.

If anyone has experience with this or can recommend a robust approach to achieving this functionality, I’d greatly appreciate your insights.

**Comment by GabenGar** [comment]
All frameworks implement their different flavour of client navigation so you literally have to rely on "specific framework features" for that. And neither it will be robust at any point.
